test/00/groupResize.sh

```diff
@@ -27,36 +27,73 @@ pass()
 }
 
 trap "fail" 1 2 3 15
+cat >input.py <<EOF
+import sys
+sys.path.insert(0, '$here')
+from pyminsky import minsky
 
-# insert ecolab script code here
-# use \$ in place of $ to refer to variable contents
-# exit 0 to indicate pass, and exit 1 to indicate failure
-cat >input.tcl <<EOF
-source $here/test/assert.tcl
-proc afterMinskyStarted {} {uplevel #0 {
- minsky.load $here/examples/GoodwinLinear02.mky
- recentreCanvas
- set item minsky.canvas.item
- assert {[findObject Group]}
- set x [\$item.right]
- set y [\$item.bottom]
- set w [\$item.width]
- set h [\$item.height]
- set z [\$item.zoomFactor]
-
- canvas.mouseDown \$x \$y
- set x [expr \$x+100]
- set y [expr \$y+100]
- canvas.mouseUp \$x  \$y
-
- findObject Group
- assert "abs(\$x-[minsky.canvas.item.right])<6"
- assert "abs(\$y-[minsky.canvas.item.bottom])<6"
- tcl_exit
-}}
+# Function to simulate assertion behavior
+def assert_condition(condition, message="Assertion failed!"):
+    if not condition:
+        raise AssertionError(message)
+
+# Generalized function to find and return the specified item or group type on the canvas
+def findObject(type_name):
```

Comment:
This method should be split into a module, as it is generically useful

Reply:
LGTM
<!-- Reviewable comment -OBF81c-1jXmM-XZsvm0:bb6ywmx -->
<!-- Sent from Reviewable.io -->